File: all-submissions/0005-longest-palindromic-substring/solution.java
class Solution {

    public boolean longestPalindromeRecur(String s, int i, int j, boolean[][] dp) {
        if(i>=j) {
            return true;
        }
        if(dp[i][j] == true) return true;
        if(s.charAt(i) == s.charAt(j)) {
            dp[i][j] = longestPalindromeRecur(s, i+1, j-1, dp);
        } else {
            dp[i][j]= false;
        }
        return dp[i][j];
    }

    public String longestPalindrome(String s) {
        String result = "";
        boolean[][] dp = new boolean[s.length()][s.length()];
        for(int i=0; i<s.length();i++){
            for(int j=i;j<s.length();j++){
                if(longestPalindromeRecur(s, i, j, dp)){
                    if(j-i+1 > result.length()){
                        result = s.substring(i, j+1);
                    }
                }
            }
        }
        return result;
    }
}

File: all-submissions/0005-longest-palindromic-substring/solution.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        n = len(s)
        dp = [[False] * n for _ in range(n)] 
        ans = [0, 0]
        for i in range(len(s)):
            dp[i][i] = True
        for i in range(n-1):
            if(s[i] == s[i+1]):
                dp[i][i+1] = True
                ans = [i, i+1]
        for l in range(2, n):
            for i in range(0, n-l):
                j = i + l
                if (s[i] == s[j] and dp[i+1][j-1]):
                    dp[i][j] = True
                    ans = [i, j]
        i, j = ans
        return s[i:j+1]
                

